### PR TITLE
Fix and improve incorrect error messages

### PR DESCRIPTION
**Pull request by delvh**

### CODE REVIEW

**Comment by CraigMason** [comment]
Hi, I think this was in response to my message in discord. This was the file that showed the error, perhaps this needs updating too:

https://github.com/go-gitea/gitea/blob/main/modules/translation/translation.go#L60

**Comment by delvh** [comment]
Yes and no:
Yes, this is in response to your Discord message.
No, they don't need to be updated because I don't see a mistake there, and those are logger notifications instead of errors, so they are intended to be viewed by humans.

File: modules/options/static.go [hunk 1]
```diff
@@ -1,4 +1,4 @@
-// Copyright 2016 The Gitea Authors. All rights reserved.
+// Copyright 2016-2022 The Gitea Authors. All rights reserved.
```
Comment (zeripath) [inline modules/options/static.go]:
```suggestion
// Copyright 2016 The Gitea Authors. All rights reserved.
```

File: modules/options/static.go [hunk 2]
```diff
@@ -32,24 +32,23 @@ func Dir(name string) ([]string, error) {
 	customDir := path.Join(setting.CustomPath, "options", name)
 	isDir, err := util.IsDir(customDir)
 	if err != nil {
-		return []string{}, fmt.Errorf("Failed to check if custom directory %s is a directory. %v", err)
+		return []string{}, fmt.Errorf("failed to check if custom directory %s is a directory. %w", customDir, err)
```
Comment (zeripath) [inline modules/options/static.go]:
```suggestion
		return []string{}, fmt.Errorf("unable to check if custom directory %q is a directory. %w", customDir, err)
```

File: modules/options/static.go [hunk 3]
```diff
@@ -32,24 +32,23 @@ func Dir(name string) ([]string, error) {
...
 	}
 	if isDir {
 		files, err := util.StatDir(customDir, true)
 		if err != nil {
-			return []string{}, fmt.Errorf("Failed to read custom directory. %v", err)
+			return []string{}, fmt.Errorf("failed to read custom directory %s. %w", customDir, err)
```
Comment (zeripath) [inline modules/options/static.go]:
```suggestion
			return []string{}, fmt.Errorf("unable to read custom directory %q. %w", customDir, err)
```

File: modules/options/static.go [hunk 4]
```diff
@@ -32,24 +32,23 @@ func Dir(name string) ([]string, error) {
...
 		}
 
 		result = append(result, files...)
 	}
 
 	files, err := AssetDir(name)
 	if err != nil {
-		return []string{}, fmt.Errorf("Failed to read embedded directory. %v", err)
+		return []string{}, fmt.Errorf("failed to read embedded directory %s. %w", name, err)
```
Comment (zeripath) [inline modules/options/static.go]:
```suggestion
		return []string{}, fmt.Errorf("unable to read embedded directory %q. %w", name, err)
```

File: modules/options/static.go (hunk 1 above)
Comment (delvh) [reply modules/options/static.go]: How did we handle it when updating existing files?
Did we really not update the copyright?
Or did we simply replace it with the current year?
```suggestion
// Copyright 2022 The Gitea Authors. All rights reserved.
```

**Review by zeripath — APPROVED** [review]

**Pull request merged**